src/theory/fp/fp_converter.cpp

```diff
@@ -12,25 +12,1029 @@
  ** \brief Conversion of floating-point operations to bit-vectors using symfpu.
  **/
 
+// Only needed for the leaf test
+#include "theory/theory.h"
+
 #include "theory/fp/fp_converter.h"
 
 #include <stack>
 
+#include "symfpu/core/packing.h"
+#include "symfpu/core/sign.h"
+#include "symfpu/core/add.h"
+#include "symfpu/core/multiply.h"
+#include "symfpu/core/fma.h"
+#include "symfpu/core/sqrt.h"
+#include "symfpu/core/divide.h"
+#include "symfpu/core/compare.h"
+#include "symfpu/core/classify.h"
+#include "symfpu/core/convert.h"
+#include "symfpu/core/remainder.h"
+
+
+namespace symfpu {
+  using namespace ::CVC4::theory::fp::symfpuSymbolic;
+
+#define CVC4SYMITEDFN(T) template <>					\
+  struct ite<symbolicProposition, T> { \
+    static const T iteOp (const symbolicProposition &_cond,	\
+			  const T &_l,					\
+			  const T &_r) {				\
+      ::CVC4::NodeManager* nm = ::CVC4::NodeManager::currentNM();	\
+      									\
+      ::CVC4::Node cond = _cond.getNode();				\
+      ::CVC4::Node l = _l.getNode();					\
+      ::CVC4::Node r = _r.getNode();					\
+									\
+      /* Handle some common symfpu idioms */				\
+      if (cond.isConst()) {						\
+	if (cond == nm->mkConst(::CVC4::BitVector(1U, 1U))) {		\
+	  return l;							\
+	} else {							\
+	  return r;							\
+	}								\
+      } else {								\
+	if (l.getKind() == ::CVC4::kind::BITVECTOR_ITE) {		\
+	  if (l[1] == r) {						\
+	    return nm->mkNode(::CVC4::kind::BITVECTOR_ITE,		\
+			      nm->mkNode(::CVC4::kind::BITVECTOR_AND,	\
+					 cond,				\
+					 nm->mkNode(::CVC4::kind::BITVECTOR_NOT, l[0])), \
+			      l[2],					\
+			      r);					\
+	  } else if (l[2] == r) {					\
+	    return nm->mkNode(::CVC4::kind::BITVECTOR_ITE,		\
+			      nm->mkNode(::CVC4::kind::BITVECTOR_AND,	\
+					 cond,				\
+					 l[0]),				\
+			      l[1],					\
+			      r);					\
+	  }								\
+	  								\
+	} else if (r.getKind() == ::CVC4::kind::BITVECTOR_ITE) {	\
+	  if (r[1] == l) {						\
+	    return nm->mkNode(::CVC4::kind::BITVECTOR_ITE,		\
+			      nm->mkNode(::CVC4::kind::BITVECTOR_AND,	\
+					 nm->mkNode(::CVC4::kind::BITVECTOR_NOT, cond), \
+					 nm->mkNode(::CVC4::kind::BITVECTOR_NOT, r[0])), \
+			      r[2],					\
+			      l);					\
+	  } else if (r[2] == l) {					\
+	    return nm->mkNode(::CVC4::kind::BITVECTOR_ITE,		\
+			      nm->mkNode(::CVC4::kind::BITVECTOR_AND,	\
+					 nm->mkNode(::CVC4::kind::BITVECTOR_NOT, cond), \
+					 r[0]),				\
+			      r[1],					\
+			      l);					\
+	  }								\
+	  								\
+	}								\
+      }									\
+      return T(::CVC4::NodeManager::currentNM()->mkNode(::CVC4::kind::BITVECTOR_ITE, cond, l, r)); \
```

Comment:
`return T(nm->mkNode(::CVC4::kind::BITVECTOR_ITE, cond, l, r));`

Reply:
I would love to say this would teach me to not just copy and paste but to update as well but ... I don't think it will :-\